code.js

```diff
@@ -1,3 +1,52 @@
-function permutationSort(a) {
-    return 0;
+// Help was once again taken from the repo of ClaytonBrown4741 to 
+// understand the structure of each permutation, and the way to
+// count the permutations was a great idea!
+
+function permutationCheck(arr, currPos) 
+{
+  let check = false;
+  for (let i = currPos; i < arr.length; i++) 
+  {
+    swap(arr, i, currPos);
+    check = permutationCheck(arr, currPos + 1);
+    if (check == true) 
+    {
+      return true;
+    }
+    swap(arr, currPos, i);
+  }
+  if (currPos >= arr.length) 
+  {
+    perm[0]++;
+    return sortCheck(arr);
+  }
+}
+
+function sortCheck(arr) 
+{
+  for (let i = 0; i < arr.length - 1; i++) 
+  {
+    if (arr[i] > arr[i + 1] || arr[i] < arr[i - 1]) 
+    {
+      return false;
+    }
+  }
+  return true;
+}
+
+function swap(arr, first, second) 
+{
+  let tmp = arr[first];
+  arr[first] = arr[second];
+  arr[second] = tmp;
+}
+
+let perm = [0];
```

Comment:
Why do you need a global variable here? Doesn't this lead to problems if you sort multiple lists?

Reply:
The perm = [0] was actually an idea I got from clayton because I was running into the problem of trying to get the other functions to recognize the perm counter and when I tried to pass it around it got messy, so when I saw clayton's idea of doing it outside and making it an array, it worked. I tried to declare var perm = [0] inside the permutationSort function and that didn't work either, so I'm not sure how to count permutations without the global variable :/

Reply:
I think I fixed it to not be a global variable!